vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -1460,13 +1480,20 @@ def process_chunk(chunk_start, chunk_end, skip_result_store=False):
     def forward_impl(self, hidden_states: torch.Tensor,
                      router_logits: torch.Tensor):
         assert self.quant_method is not None
+        # Route to the chunked forward path using the FlashInfer Cutlass kernel
+        # only when data parallelism (DP) is enabled.
```

Comment:
So for TP we can still use the FlashInfer Cutlass path, but with no need for chunking?